cmd/createorupdate/createorupdate.go

```diff
@@ -1,25 +1,32 @@
 package main
 
 import (
-	"bytes"
 	"context"
+	"encoding/json"
 	"io/ioutil"
 	"net/http"
 	"os"
+	"sync"
 
 	"github.com/ghodss/yaml"
 	"github.com/sirupsen/logrus"
 
 	"github.com/openshift/openshift-azure/pkg/api"
 	v20180930preview "github.com/openshift/openshift-azure/pkg/api/2018-09-30-preview/api"
 	"github.com/openshift/openshift-azure/pkg/fakerp"
+	"github.com/openshift/openshift-azure/pkg/util/azureclient"
+	sdk "github.com/openshift/openshift-azure/pkg/util/azureclient/osa-go-sdk/services/containerservice/mgmt/2018-09-30-preview/containerservice"
 )
 
 type server struct {
 	// the server will not process more than a single
-	// request at all times.
+	// PUT request at all times.
 	inProgress chan struct{}
-	log        *logrus.Entry
+
+	sync.RWMutex
+	oc *v20180930preview.OpenShiftManagedCluster
```

Comment:
Will this not cache the previous request into the server? If I send `PUT` with `cluster1` and get succ, and later `PUT` with `cluster2` it will get `cluster1` value from cache?

Reply:
Otherwise I need to restart server with each new set of requests?

Reply:
Maybe cache should be done with something like https://github.com/golang/groupcache and add the ability to have multiple clusters/request in the same running server? We could host our fake RP outside and do more realistic testing?

Reply:
I don't think we are going to support deploying multiple clusters in our fake RP. Why would we? Anyway, it's out of scope for this PR. As far as this field is concerned, I think we can probably hold just the provisioning state.

Reply:
Actually we also need to hold oc in order to reply to GET requests with it and the provisioning state.